File: crates/fixtures/sample.ts
// Sample TypeScript module for testing
import * as fs from 'fs';

/**
 * A helper function.
 */
function helper(x: number): number {
    return x * 2;
}

function process(items: number[]): number {
    // HACK: naive loop
    let total = 0;
    for (const item of items) {
        total += helper(item);
    }
    return total;
}

export { helper, process };

File: crates/fixtures/sample.py
"""Sample Python module for testing."""

import os
import sys
from typing import List

def helper(x: int) -> int:
    """A helper function."""
    return x * 2

def process(items: List[int]) -> int:
    # TODO: add error handling
    total = 0
    for item in items:
        total += helper(item)
    return total

def unused(a: str, b: str) -> str:
    return a + b
